framework/log/FileTarget.php

```diff
@@ -163,7 +165,7 @@ protected function rotateFiles()
                     continue;
                 }
                 $newFile = $this->logFile . '.' . ($i + 1);
-                $this->rotateByCopy ? $this->rotateByCopy($rotateFile, $newFile) : $this->rotateByRename($rotateFile, $newFile);
+                $this->rotateByCopy($rotateFile, $newFile);
```

Comment:
@bizley Did you also test this on Windows? It uses mandatory locking with `flock()` and I'm not sure if a locked file can be  `unlink()`'ed. A hacky solution could be to release the lock in `rotateFiles()` right before the `unlink()` happens. I'm not sure though.

Reply:
I did not :/ Hoped that it's covered by the pipe. As I read it seems to be the case that Windows is not deleting the file until the last handle is released but I'm not sure if it throws an error or not. Anyway, question is should I switch back the line for rotating back where it was (after the unlocking) or something else needs to be done.

Reply:
Moving the line back leaves us with the initial problem: Lock is released too early and while we start rotating in this process a second process can try to start log handling in parallel.

Maybe @stu2162583 (or someone else with PHP on Windows) can help out and do a quick test if a locked file can be deleted under windows from the same process?

I don't know of any other solution - except maybe using a mutex component and not use any file locks here at all.

Reply:
Quick look at the Monolog - the file is locked only for writing, rotating is done before the writing, for the very moment before the unlink they check whether the file is writable and if so they disable the error handler, unlink the file, and enable handler again.

Reply:
Seems like they don't use any locking at all - but their whole log handling logic might also be quite different from ours. I don't know.

Still think that we should keep some sort of locking.